### ISSUE TITLE
[Export] Implement TorchScript model export

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#7]
## Goal

Add code to export the trained model to TorchScript (`.pt`) for deployment outside of Python/PyTorch Lightning.

## Requirements (from spec Milestone 4)

- Use `torch.jit.script()` or `torch.jit.trace()` to export the model
- The exported model should accept a preprocessed image tensor and return the 4-head logits dict
- Include a short usage example showing how to load and run the exported model
- Save as `model.pt` (or a timestamped path)

## Suggested interface

```python
export_torchscript(model, output_path="model.pt", example_input=None)
```

## Notes

- `torch.jit.trace` is simpler but requires an example input; `torch.jit.script` is more robust but requires type annotations
- The dictionary return type from `forward()` must be typed correctly for TorchScript compatibility

## Files
- `src/models/multi_head_resnet.py` (or `src/models/export.py`)

## Depends on
- Issue #2 (model architecture must be finalized first)
